api/core/v1beta2/common_types.go

```diff
@@ -181,6 +181,11 @@ const (
 	// This annotation can be used to inform MachinePool status during in-progress scaling scenarios.
 	ReplicasManagedByAnnotation = "cluster.x-k8s.io/replicas-managed-by"
 
+	// InPlaceUpdateInProgressAnnotation is set on Machine, InfraMachine, and BootstrapConfig when an in-place update is in progress.
```

Comment:
Fine for now, just a heads up. This diff should go away after a rebase after https://github.com/kubernetes-sigs/cluster-api/pull/12865 is merged